src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -30,6 +30,9 @@ import sifive.blocks.util.{DeviceParams,DeviceAttachParams}
 
 class GPIOPortIO(val c: GPIOParams) extends Bundle {
   val pins = Vec(c.width, new EnhancedPin())
+}
+
+class IOFPortIO(val c: GPIOParams) extends Bundle {
```

Comment:
Since this Bundle doesn't really have much to do with the GPIO device at all anymore, can you make it just directly take the parameters it needs vs taking an entire GPIOParams?

Reply:
OK. I agree with you. I will fix it.

Reply:
You mean both GPIOPortIO and IOFPortIO take a int parameter for width. Is this right?

Reply:
i would not change `GPIOPortIO` (since that will be a bigger API change). Just make IOFPortIO take the params it needs.

Reply:
or alternatively call this "GPIOIOFPortIO" or something that links it to the GPIO device

Reply:
but i think the former suggestion is more sensical

Reply:
I don't want to make big changes for this PR. Because, we plan to release chronos feature complete RTL.
I will change parameter of IOFPortIO, only.